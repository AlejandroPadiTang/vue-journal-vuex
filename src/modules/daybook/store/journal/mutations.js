
export const setEntries = ( state, entries ) => {
    state.entries = [...state.entries, ...entries];
    state.isLoading = false;
}

export const updateEntry = ( state, entry ) => {
    const idx = state.entries.findIndex(e => e.id === entry.id);
    state.entries[idx] = entry;
}

export const addEntry = ( state, entry ) => {
    state.entries = [...state.entries, entry];
}

export const deleteEntry = ( state, id ) => {
    // se puede usar filter?
    const idx = state.entries.findIndex(e => e.id === id);
    state.entries.splice(idx, 1);
}