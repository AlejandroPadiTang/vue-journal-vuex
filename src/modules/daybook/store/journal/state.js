export default () => ({
    isLoading: true,
    entries: [
        {
            id: new Date().getTime(),
            date: new Date().toDateString(),
            text: 'Lorem ipsum dolor sit amet consectetur adipisicing elit. Ipsum similique nam architecto tempore perferendis quia culpa quae distinctio explicabo error facilis porro iusto praesentium, aut tempora ab laudantium consequatur molestias.',
            picture: null
        },
        {
            id: new Date().getTime() + 1000,
            date: new Date().toDateString(),
            text: 'nam architecto tempore perferendis quia culpa quae distinctio explicabo error facilis porro iusto praesentium, aut tempora ab laudantium consequatur molestias.',
            picture: null
        },
        {
            id: new Date().getTime() + 3000,
            date: new Date().toDateString(),
            text: 'quia culpa quae distinctio explicabo error facilis porro iusto praesentium, aut tempora ab laudantium consequatur molestias.',
            picture: null
        }
    ]
})