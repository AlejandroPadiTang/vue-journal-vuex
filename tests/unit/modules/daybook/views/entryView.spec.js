/* eslint-disable no-unused-vars */
import { shallowMount } from '@vue/test-utils';
import { createStore } from 'vuex';
import journal from '@/modules/daybook/store/journal';
import EntryView from '@/modules/daybook/views/EntryView.vue';
import { journalState } from '../../../mocks/test-journal-state';

import Swal from 'sweetalert2';

const createVuexStore = ( initialState ) =>
  createStore({
    modules: {
      journal: {
        ...journal,
        state: { ...initialState }
      }
    }
  });

  jest.mock('sweetalert2', () => ({
    fire: jest.fn(),
    showLoading: jest.fn(),
    close: jest.fn()
  }));

describe('Pruebas en EntryView', () => {
  const store = createVuexStore(journalState);
  store.dispatch = jest.fn();

  const mockRouter = {
    push: jest.fn()
  }

  let wrapper;
  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallowMount(EntryView, {
      props: {
        id: journalState.entries[0].id
      },
      global: {
        mocks: {
          $router: mockRouter
        },
        plugins: [ store ]
      }
    })
  })

  test('debe de sacar al usuario porque el id no existe', () => {
    const wrapper = shallowMount(EntryView, {
      props: {
        id: 'Este ID no existe en el store'
      },
      global: {
        mocks: {
          $router: mockRouter
        },
        plugins: [ store ]
      }
    });

    expect( mockRouter.push ).toHaveBeenCalledWith({ name: 'no-entry' });
  });

  test('debe de mostrar la entrada correctamente', () => {
    expect(wrapper.html()).toMatchSnapshot();
    expect(mockRouter.push).not.toHaveBeenCalled();
  });

  test('debe de borrar la entrada y salida', (done) => {
    Swal.fire.mockReturnValueOnce( Promise.resolve({ isConfirmed: true }));
    const btn = wrapper.find('[data-testid="btn-delete"]');
    btn.trigger('click');
    expect( Swal.fire ).toHaveBeenCalledWith({
        title: '¿Está seguro?',
        text: 'Una vez borrado, no se puede recuperar',
        showDenyButton: true,
        confirmButtonText: 'Si, estoy seguro'
    })

    setTimeout(() => {
      expect( store.dispatch ).toHaveBeenCalledWith('journal/deleteEntry', '-NQ6_sms5sfHWpGpyrpS');
      expect( mockRouter.push ).toHaveBeenCalled();
      done();
    }, 1)
  });
})