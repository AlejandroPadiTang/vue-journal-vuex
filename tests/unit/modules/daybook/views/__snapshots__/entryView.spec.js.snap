// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en EntryView debe de mostrar la entrada correctamente 1`] = `
<div class="entry-title d-flex justify-content-between p-2">
  <div><span class="text-success fs-3 fw-bold">9</span><span class="mx-1 fs-3">Marzo</span><span class="mx-2 fs-4 fw-ligth">2023, Jueves</span></div>
  <div><input type="file" accept="image/png, image/jpeg" style="display: none;"><button class="btn btn-danger mx-2" data-testid="btn-delete"> Borrar <i class="fa fa-trash-alt"></i></button><button class="btn btn-primary"> Subir foto <i class="fa fa-upload"></i></button></div>
</div>
<hr>
<div class="d-flex flex-column px-3 h-75"><textarea placeholder="¿Qué sucedió hoy?">
      </textarea></div>
<fab-stub icon="fa-floppy-disk"></fab-stub>
<!--v-if-->
<!--v-if-->
`;
